browser/src/control/jsdialog/Widget.OverflowGroup.ts

```diff
@@ -101,14 +99,31 @@ function setupOverflowMenu(
 
 	// keeps original content
 	const originalTopbar = overflowMenu.querySelectorAll(':scope > *');
-
+	const bottomGroup = parentContainer.querySelector(
+		'.ui-overflow-group-bottom',
+	) as HTMLElement;
 	// hide/show content on resize
 	const overflowMenuHandler = (hideContent: boolean) => {
 		overflowMenu.replaceChildren();
+		hiddenItems.replaceChildren();
+
+		// Create a new container for the top row
+		const topRow = document.createElement('div');
+		topRow.classList.add('top-row-overflow-group');
+
 		originalTopbar.forEach((element: Element) => {
-			overflowMenu.append(element);
+			topRow.append(element);
 		});
 
+		overflowMenu.append(topRow);
+
+		if (bottomGroup && hideContent) {
+			overflowMenu.append(bottomGroup);
+		} else {
+			const parent = overflowMenu.parentNode as HTMLElement;
+			const grandParent = parent?.parentNode as HTMLElement;
+			grandParent?.insertBefore(bottomGroup, parent.nextSibling);
```

Comment:
Could we explicitly target the nodes? I mean we have these "container", "contentContainer", "topBar" etc. names. It would be easier to follow by using these references than "parent".

(I assume it would be useful to also rename parameters of `setupOverflowMenu` function so they match the names as we create them in the main function to reduce complexity.